feat: Winston

kimseungbin opened the pull request:
# Description

- 아직 작업중입니다. Config 및 infra 설정이 완료되어야 더 진행이 가능합니다.
- 추가적인 개발 방향에 대해 논의가 필요합니다. 모두 10/20 일 금요일 오전 중으로 최소한의 의견 혹은 커밋을 보태주셨으면 합니다.

# Key Changes

- Winston 및 관련 dependencies 설치
- Nest default logger -> Winston 대체

# Issue Reference
- Close #58 

# Package Change

# Checklist

- [x] Asignee를 본인으로 지정하셨나요?
- [x] Reviewer를 2명 이상 지정 했나요?
- [ ] 코드의 엣지 케이스에 대해 충분히 고민 하셨나요?
- [x] conflict가 나지 않는지 확인 하셨나요?

kimseungbin committed:
chore: install winston, winston-cloudwatch, and nest-winston

kimseungbin committed:
feat: implement Winston logger including nest bootstrapping

Uses Winston's default format
cf. https://github.com/winstonjs/winston#formats

Ooyd commented:
기본적으로 사용되어지는 옵션값에 대해 찾아봤습니다.

winston log

format은 json형식
transports.console은 timestamp와 colorize만 우선적으로 설정해두면 좋을거같아요.
cloudwatchConfig

Cloudwatch Group명
logStream 명(일반적으로 고유명)
API인증 Access키
API인증 SecretAccess키
awsRegion키
messageFormatter 형식값
[Winston and Cloudwatch set up](https://javascript.plainenglish.io/set-up-a-logger-for-your-node-app-with-winston-and-cloudwatch-in-5-minutes-dec0c6c0d5b8)

kimseungbin commented:
@Ooyd 일단 색상과 JSON 화는 로컬에서만 적용하자는 말씀이시죠?

kimseungbin commented:
@Ooyd 
![image](https://github.com/HPlus1100/server/assets/24871719/f47ed99b-7e98-44be-8651-1ebb02040806)
이런 느낌을 원하시는게 맞으실까요?

kimseungbin committed:
feat: add consoleTransportOptions for development env

It adds colors and timestamps for local development.
Also added format.simple() to all envs

Ooyd commented:
> @Ooyd 일단 색상과 JSON 화는 로컬에서만 적용하자는 말씀이시죠?

네 각자 개발환경에서도 보기좋게 확인이 되면 좋을거같습니다.

Ooyd commented on src/config/app.config.ts:
```diff
@@ -0,0 +1,20 @@
+import { registerAs } from '@nestjs/config';
+import * as process from 'process';
```
process명시적으로 한 이유가 있을까요? 
단순한 궁금증입니다!

kimseungbin replied:
아.. IDE 가 스스로 해버렸네요.
사실 있어도 문제는 되지 않아서 남겨놔도 괜찮을까요?
(놓친 부분입니다.)
